converter/output/rita/db.go

```diff
@@ -0,0 +1,140 @@
+package rita
+
+import (
+	"github.com/activecm/ipfix-rita/converter/output/rita/buffered"
+	"github.com/activecm/ipfix-rita/converter/output/rita/constants"
+	"github.com/activecm/ipfix-rita/converter/output/rita/freqconn"
+	"github.com/activecm/rita/parser/parsetypes"
+	"github.com/globalsign/mgo"
+	"github.com/pkg/errors"
+	"time"
+)
+
+//DB represents a RITA database and provides routines
+//for constructing a new RITA database and inserting
+//new data into that database
+type DB struct {
+	manager         DBManager
+	outputDB        *mgo.Database
+	connColl        *buffered.AutoFlushCollection
+	connCounter     freqconn.ConnCounter
+	strobesNotifier freqconn.StrobesNotifier
+}
+
+func newDB(dbManager DBManager, outputDB *mgo.Database,
+	strobeThreshold int, bufferSize int64, flushDeadline time.Duration,
+	asyncErrorChan chan<- error, onFatalError func()) (DB, error) {
+
+	strobesSess := outputDB.Session.Copy()
```

Comment:
The AutoFlushCollection automatically closed the underlying session when it was closed. It was easier to model the strobesNotifier using this pattern than change the old code. So, here we create new sessions for the strobesNotifier and the conn autoflushcollection.